plugins/woocommerce/src/Blocks/Templates/AbstractTemplatePart.php

```diff
@@ -0,0 +1,58 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Templates;
+
+use Automattic\WooCommerce\Blocks\BlockTemplatesRegistry;
+use Automattic\WooCommerce\Blocks\Utils\BlockTemplateUtils;
+
+/**
+ * AbstractTemplatePart class.
+ *
+ * Shared logic for templates parts.
+ *
+ * @internal
+ */
+abstract class AbstractTemplatePart {
```

Comment:
There's a lot of duplicate code between this and the `AbstractTemplate` class, can we not use one class for both, or extend the `AbstractTemplate` class in here?

Reply:
I initially only create the `AbstractTemplate` class, but decided to split it in two because:

* Template parts need to have `$template_area`, while regular templates don't.
* In the constructor we need to check if the theme supports `wp_template` or `wp_template_part`.

In any case, yeah, it should be possible to make `AbstractTemplatePart` to extend `AbstractTemplate`. I implemented it in: b53dc771d652578d2f99722f28e407b7b3544584.